iaso/institutions/wikidataclient.py

```diff
@@ -13,10 +13,11 @@
 class WikiDataClient:
     def __init__(self):
         self.client = httpx.AsyncClient(
-            timeout=httpx.Timeout(HTTPX_TIMEOUT, pool_timeout=None)
+            timeout=httpx.Timeout(HTTPX_TIMEOUT, pool_timeout=None),
         )
 
         self.waiting = defaultdict(int)  # Number of waiting requests
+        self.running = defaultdict(int)  # Number of running requests
         self.queue = defaultdict(list)  # Requests not yet scheduled
         self.backoff = dict()  # Exponential backoff times
```

Comment:
Is there a re-try pattern offered by any library in a way that Spring offers it for Java?

Reply:
For httpx which we are using here, there unfortunately is not (see https://github.com/encode/httpx/pull/134#issuecomment-515954596). There is one for the lower level urrlib3 (see https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#urllib3.util.retry.Retry) which we do not have direct access to here. We could also use a higher level retrying library like https://github.com/jd/tenacity. However, this class is not only responsible for retrying but also for scheduling all requests so we do not overload the API endpoint. I have not yet seen this implemented in a retry library as well.